shanoir-ng-studies/src/main/java/org/shanoir/ng/study/dua/DataUserAgreementService.java

```diff
@@ -44,32 +45,28 @@
 	}
 	
 	public void acceptDataUserAgreement(Long duaId) throws ShanoirException {
+		Long userId = KeycloakUtil.getTokenUserId();
 		DataUserAgreement dataUserAgreement = repository.findOne(duaId);
-		dataUserAgreement.setTimestampOfAccepted(new Date());
-		repository.save(dataUserAgreement);
-		StudyUser studyUser = repositoryStudyUser.findByUserIdAndStudy_Id(dataUserAgreement.getUserId(), dataUserAgreement.getStudy().getId());
-		if (studyUser == null) {
-			throw new ShanoirException("Could not validate the data user agreement acceptation. The user don't seems to be a member of the study.");
-		}
-		studyUser.setConfirmed(true);
-		repositoryStudyUser.save(studyUser);
-		// Send updates via RabbitMQ
-		try {
-			List<StudyUserCommand> commands = new ArrayList<>();
-			commands.add(new StudyUserCommand(CommandType.UPDATE, studyUser));
-			studyUserCom.broadcast(commands);
-		} catch (MicroServiceCommunicationException e) {
-			LOG.error("Could not transmit study-user update info through RabbitMQ");
-		}
-	}
-	
-	public void createDataUserAgreementsForStudy(Study study) {
-		List<StudyUser> studyUserList = study.getStudyUserList();
-		for (StudyUser studyUser : studyUserList) {
-			DataUserAgreement dataUserAgreement = new DataUserAgreement();
-			dataUserAgreement.setStudy(study);
-			dataUserAgreement.setUserId(studyUser.getUserId());
+		// assure that only the user itself can accept its DUA
+		if (dataUserAgreement.getUserId().compareTo(userId) == 0) {
```

Comment:
Why not using "equals" for code clarity ? (change it only if using @PreAuthorize as Julien was stating).

We could add some unit tests at the same time